sql/src/main/java/io/crate/execution/jobs/RootTask.java

```diff
@@ -266,6 +266,9 @@ private void finish() {
             } catch (Throwable t) {
                 profilingFuture.completeExceptionally(t);
             }
+            if (failure != null) {
+                finishedFuture.completeExceptionally(failure);
```

Comment:
Would it make sense to do this https://github.com/crate/crate/blob/5a300ae028ec050863ea1603f308fe6ba2407f3e/sql/src/main/java/io/crate/execution/jobs/RootTask.java#L202-L204 here already? Or to chain the finishedFuture / profilingFuture in another way, so that if one is set the other is also triggered?

Reply:
`finishedFuture.complete(..)` is here https://github.com/crate/crate/blob/5a300ae028ec050863ea1603f308fe6ba2407f3e/sql/src/main/java/io/crate/execution/jobs/RootTask.java#L282

but it seems that it is just for the success scenario, so they are kind of chained?

I'll look at `start` part now.

Reply:
Maybe the problem is that in the early failure case `finishProfiling` is never called, so the chaining of the futures never happens. `close` would also take care of setting a failure.

Reply:
as discussed, will merge it for now